packages/@aws-cdk/aws-ec2/README.md

```diff
@@ -88,6 +88,53 @@ itself to 2 Availability Zones.
 Therefore, to get the VPC to spread over 3 or more availability zones, you
 must specify the environment where the stack will be deployed.
 
+### Choosing subnets for resources
+
+When creating some resources, such as databases or instances, there is an option 
+to choose which subnets to place them in. For example, a VPC endpoint by default 
+is placed into a subnet in every availability zone, but you can override which 
+subnets to use.
+
+The example below will place the endpoint into two AZs (us-east-1a and us-east-1c),
+in Isolated subnets.
+```ts
+new InterfaceVpcEndpoint(stack, "VPC Endpoint", {
+  vpc,
+  service: new InterfaceVpcEndpointService("com.amazonaws.vpce.us-east-1.vpce-svc-uuddlrlrbastrtsvc", 443),
+  subnets: {
+    subnetType: SubnetType.ISOLATED,
+    availabilityZones: ["us-east-1a", "us-east-1c"]
+  }
+});
+```
+
+You can also specify particular subnets for granular control.
+```ts
+new InterfaceVpcEndpoint(stack, "VPC Endpoint", {
+  vpc,
+  service: new InterfaceVpcEndpointService("com.amazonaws.vpce.us-east-1.vpce-svc-uuddlrlrbastrtsvc", 443),
+  subnets: {
+    subnets: [subnet1, subnet2]
+  }
+});
+```
+
+Which subnets are selected is evaluated as follows:
+
+* Specifying particular subnets overrides any other settings, such as type or 
+availability zone.
+* If no subnets are specified, but a subnetGroupName is, the
```

Comment:
Thanks for documenting the rules! 

This is in fact not how it's intended. `subnetType` and `subnetGroupName` are supposed to live at the same level of precedence, where `subnetGroupName` is necessary if there are multiple *groups* with the same *type*. (subnetGroup implies subnetType). 

I think `availabilityZones` should work as an additional restrictor applied at the end, similar to when `onePerAz` is applied. So the precedence should work like this:

* Specific subnets, if given. 
* A set of subnet (groups), implied by either subnetType, subnetGroupName, or the absence of both.
  * A subset of those subnets (hah), determined by both `availabilityZones` and/or `onePerAz`.

Reply:
I see what you mean. I wrote the readme as a direct reflection of the code that does the selection ([link](https://github.com/aws/aws-cdk/blob/87b16350ed2ccea2cda03e3515cd8797c8047186/packages/%40aws-cdk/aws-ec2/lib/vpc.ts#L386-L405)) but I like yours better.

Reply:
I've updated the readme accordingly, and also modified the selection code such that it reflects that selection is primarily by group name or type, then with additional filtering by availability zone and one per AZ if specified.